src/player_input.py

```diff
@@ -10,54 +10,49 @@
 
 from __future__ import annotations
 from tokenize import String
+from typing import Optional
 import arcade
 from pyglet.input import Controller
 from pyglet.window.key import KeyStateHandler
 
 
 class PlayerInput:
-
-    # Left stick
-    x_axis: VirtualAxis
-    y_axis: VirtualAxis
-
-    # Right stick
-    rx_axis: VirtualAxis
-    ry_axis: VirtualAxis
-
-    # Buttons
-    accelerate_button: VirtualButton
-    brake_button: VirtualButton
-    primary_fire_button: VirtualButton
-    secondary_fire_button: VirtualButton
-
     def __init__(self, keys: KeyStateHandler, controller: Controller) -> None:
+        # Left stick
         self.x_axis = VirtualAxis(keys, controller)
         self.y_axis = VirtualAxis(keys, controller)
+
+        # Right stick
         self.rx_axis = VirtualAxis(keys, controller)
         self.ry_axis = VirtualAxis(keys, controller)
-        self.accelerate_button = VirtualButton(keys, controller)
-        self.brake_button = VirtualButton(keys, controller)
+
+        # Accelerate/brake
+        self.accelerate_axis = VirtualAxis(keys, controller)
+        self.brake_axis = VirtualAxis(keys, controller)
+
+        # Buttons
         self.primary_fire_button = VirtualButton(keys, controller)
         self.secondary_fire_button = VirtualButton(keys, controller)
 
+    def update(self):
+        self.primary_fire_button._update()
+        self.secondary_fire_button._update()
 
-class VirtualAxis:
-    key_negative: str = None
-    key_positive: str = None
-    button_positive: int = None
-    button_negative: int = None
-    axis: str = None
-
-    value: float
 
+class VirtualAxis:
     __keys: KeyStateHandler = None
     __controller: Controller = None
 
     def __init__(self, keys: KeyStateHandler, controller: Controller):
         self.__keys = keys
         self.__controller = controller
 
+        self.key_negative: Optional[str] = None
+        self.key_positive: Optional[str] = None
+        self.button_positive: Optional[int] = None
+        self.button_negative: Optional[int] = None
+        self.axis: Optional[str] = None
```

Comment:
`Optional[some_type]` is the way to indicate something might be `some_type` or it might be `None`.  Appropriate here because I use `None` to indicate when a key or button is unbound.